src/server/routes/entity/entity.js

```diff
@@ -482,9 +488,9 @@ async function getNextAnnotation(
 		id && new Annotation({id}).fetch({transacting})
 	);
 
-	return orm.func.annotation.updateAnnotation(
+	return body.annotation ? orm.func.annotation.updateAnnotation(
```

Comment:
if body.annotation was undefined, and when oldAnnotation was also undefined, old==new and it was returning old